easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -116,6 +117,17 @@ def build_type(self):
             build_type = 'Debug' if self.toolchain.options.get('debug', None) else 'Release'
         return build_type
 
+    def prepend_config_opts(self, config_opts):
+        """Prepends configure options (-Dkey=value) to configopts ignoring those already set"""
+        cfg_configopts = self.cfg['configopts']
+        # All options are of the form '-D<key>=<value>'
+        new_opts = ' '.join('-D%s=%s' % (key, value) for key, value in config_opts.items()
+                            if '-D%s=' % key not in cfg_configopts)
+        if new_opts:
+            if cfg_configopts:
+                new_opts += ' '
+            self.cfg['configopts'] = new_opts + ' ' + cfg_configopts
```

Comment:
Why add the extra space?

Reply:
Other than that, the code looks ok to me

Reply:
Perhaps a ' '.join([new_opts, cfg_configopts]) would be simpler

Reply:
Good catch, done.